code/__HELPERS/sorts/helpers.dm

```diff
@@ -0,0 +1,97 @@
+/// Sorts the list in place with timSort, default settings.
+#define SORT_TIM(to_sort, associative) \
+	if (length(to_sort) < 2) { \
+		return to_sort; \
+	} \
+	var/datum/sort_instance/sorter = GLOB.sortInstance; \
+	if (isnull(sorter)) { \
+		sorter = new; \
+	} \
+	sorter.L = to_sort; \
+	sorter.cmp = GLOBAL_PROC_REF(cmp_numeric_asc); \
+	sorter.associative = associative; \
+	sorter.timSort(1, 0); \
+
+
+/// Helper for the sorting procs. Prevents some code duplication. Creates /datum/sort_instance/sorter
+#define CREATE_SORT_INSTANCE(to_sort, cmp, associative, fromIndex, toIndex) \
+	if (length(to_sort) < 2) { \
+		return to_sort; \
```

Comment:
Same

Reply:
Does it make sense here, considering it's later undefined? None of the procs continue if its sorted, and making the define hygienic (by not altering from/to) would defeat the purpose of using the define